### ISSUE TITLE
Add attribute keep_case for NifUnitEnum

### ISSUE THREAD

**Issue by evnu** [rusterlium/rustler#263]
`NifUnitEnum` currently expects atoms corresponding to struct fields to be [in snake case](https://github.com/rusterlium/rustler/blob/master/rustler_codegen/src/unit_enum.rs#L74). This works fine when the user can define the atoms to be in snake case. If that is not possible, e.g. because they are defined by a third party, then decoding them with `NifUnitEnum` is not possible.

I suggest we add an attribute `keep_case` which defines that the fields are to be used exactly as-is as atoms.

```rust
#[derive(NifUnitEnum)]
#[rustler(keep_case)]
pub struct KeepCase {
  UPPERCASE,
  lowercase
}
```

**Comment by scrogson** [comment]
What about something more similar to `serde`'s `rename`?

```rust
#[derive(NifUnitEnum)]
#[rustler(keep_case)]
pub struct KeepCase {
    #[rustler(rename = "UPPERCASE")]
    uppercase,
    lowercase
}
```

I imagine that if we end up adding `serde` support to `rustler`, we will end up just using `#[serde(rename = "UPPERCASE")]` ?

**Comment by evnu** [comment]
> What about something more similar to serde's rename?

Yes, that would work. Does serde by default rename variants at all, or does it keep them as-is except when `rename` is specified?

**Comment by filmor** [comment]
Wouldn't it make sense to instead work on getting serde_rustler in? I don't see the advantage of a custom codegen right now.

**Comment by evnu** [comment]
As this is a small change, I would do this. I am unsure how much work serde_rustler will be. Of course, rustler_serde would be the preferred way.

**Comment by scrogson** [comment]
> Of course, rustler_serde would be the preferred way.

Agreed in the context of this crate. I would assume that it wouldn't really be useful outside of rustler anyways.

**Comment by sunny-g** [comment]
Mentioned [here](https://github.com/rusterlium/rustler/issues/200#issuecomment-595349747):

Depending on how willing y'all are to change up rustler and depend directly on serde, a lot of ser/de + encoder/decoder + macro logic could be simplified and/or gotten rid of. The simplest non-breaking change however would be to just copy my `Serializer` and `Deserializer` logic and implement them directly on `Env` and `Term` respectively behind a feature flag, no real need for a separate crate.

**Comment by scrogson** [comment]
I started working on integrating this into rustler over the weekend 👍
